fulpstation/fulpstation
# Official Unofficial Fulpstation Halloween 2021 Branch [DO NOT MERGE]

## About The Pull Request

This branch will collect and compile all the various halloween costumes in preparation for October so we're not sorting through 50 PRs at the final day.

### If you want your costume added to the branch, here are your options:

1. Make a pull request to my halloween-2021 branch
2. If you don't know how to make a pull request, learn how to make one
3. If you can't learn how to make one, provide me with AT LEAST the .dmi files and I'll add them myself

### How to add your costume to the game:

1. Make 2 .dmi files, one for the icons when worn and one for the icon while in the inventory. They should be called [costume_name]_worn.dmi and [costume_name]_icon.dmi
2. Add the sprites to both of the files in separate icon states for each piece of clothing. The worn icon states should be directional, while the item icon states should not (Click the 4 arrows near the bottom left to make a sprite directional). Keep the icon state names consistent between the worn and icon files for each piece of clothing
3. Add the code, following the format of previous costumes in halloween_2021.dm by making a new object (/obj/item/clothing/[type]/[name]) and then pointing to the correct icon files and icon states using the icon, worn_icon, and icon_state variables.
4. MAKE SURE THE ICON STATE NAMES ARE IDENTICAL BETWEEN FILES, AND MAKE SURE YOU USE DOUBLE QUOTES FOR NON FILE VARIABLES AND SINGLE QUOTES FOR FILE VARIABLES
5. Make a box for your set using previous costumes as reference. At a minimum just give it a theme_name variable so people know what's in the box (/obj/item/storage/box/halloween/edition_21/[name])
6. Modify the box's PopulateContents proc as seen earlier in the file to put the items in the box
7. Run Build.bat through Visual Studio Code (By using the Ctrl+Shift+B hotkey) or File Explorer and test ingame to make sure everything is working properly

You can use the template image provided here as a base for making a costume

![template](https://user-images.githubusercontent.com/83368538/129089291-1ba45c0f-3335-4827-bda9-d50a71168d6d.png)


### How to test your costume in-game:

1. Open up tgstation.dmb after you build the code
2. Load in game
3. Go to the debug tab and hit spawn
4. Spawn this year's choice beacon (/obj/item/choice_beacon/halloween/edition_21)
5. Use the beacon to spawn your box. It should be in the list if you set it up correctly
6. Wear your costume and see how it looks
7. If your items are invisible, make sure you set up the files correctly and make sure you saved the .dmi files in dream maker. You will have to re-build in order to test your costumes again if you forget.


### Common Clothing Paths You Will Want To Use:

/obj/item/clothing/suit/[name] (For things that go in the outer chest slot, like firesuits)
/obj/item/clothing/shoes/[name] (For shoes)
/obj/item/clothing/head/[name] (For things that go on the head)
/obj/item/clothing/under/[name] (For things that go on the jumpsuit slot. You will need to set more variables if you use this to make sure your costume has sensors and cant be adjusted)
/obj/item/clothing/suit/hooded/[name] (For things that go in the outer chest slot and have a hood, like winter coats)
/obj/item/clothing/head/hooded/[name] (For things that go on your head when you pull your hood up, like a winter coat hood)

## Changelog
:cl:
add: Various Halloween Costumes
code: Made the framework for adding 2021 Halloween Costumes
/:cl:
